### PR TITLE
core/state/pruner: fix compaction range error

### PR DESCRIPTION
**Pull request by holiman**
Fix for https://github.com/ethereum/go-ethereum/pull/22291#discussion_r572263644

### CODE REVIEW

**Comment by holiman** [comment]
https://play.golang.org/p/66KgDfD-96d

File: core/state/pruner/pruner.go [hunk 1]
```diff
@@ -186,16 +186,15 @@ func prune(maindb ethdb.Database, stateBloom *stateBloom, middleStateRoots map[c
...
-				end   = []byte{(b+1)<<4 - 1}
+				start = []byte{byte(b)}
+				end   = []byte{byte(b + 0x10)}
 			)
-			log.Info("Compacting database", "range", fmt.Sprintf("%#x-%#x", start, end), "elapsed", common.PrettyDuration(time.Since(cstart)))
-			if b == 15 {
+			if b == 0xf0 {
 				end = nil
 			}
+			log.Info("Compacting database", "range", fmt.Sprintf("%#x-%#x", start, end), "elapsed", common.PrettyDuration(time.Since(cstart)))
```
Comment (karalabe) [inline core/state/pruner/pruner.go:197]: If you move it below the `if`, you'll end up printing `nil` into hex. Not sure how that behaves.

Comment (karalabe) [reply core/state/pruner/pruner.go:197]: The very last range's end.

Comment (holiman) [reply core/state/pruner/pruner.go:197]: Right, neither did I, that's why I made the play thingy above
```
0x00-0x10
0x10-0x20
0x20-0x30
0x30-0x40
0x40-0x50
0x50-0x60
0x60-0x70
0x70-0x80
0x80-0x90
0x90-0xa0
0xa0-0xb0
0xb0-0xc0
0xc0-0xd0
0xd0-0xe0
0xe0-0xf0
0xf0-
```

Comment (karalabe) [reply core/state/pruner/pruner.go:197]: Funky

**Pull request merged**